src/main/java/org/fcrepo/auth/webac/WebACRolesProvider.java

```diff
@@ -387,16 +357,14 @@ private FedoraResource getBaseVersion(final FedoraResource resource) {
             resource.getChildren().forEach(child -> {
                 if (child.getTypes().contains(WEBAC_AUTHORIZATION)) {
                     final Map<String, List<String>> aclTriples = new HashMap<>();
-                    child.getTriples(translator, PROPERTIES)
-                        .filter(p -> isAclPredicate.test(model.asStatement(p).getPredicate()))
-                        .forEach(t -> {
-                            aclTriples.putIfAbsent(t.getPredicate().getURI(), new ArrayList<>());
-                            if (t.getObject().isURI()) {
-                                aclTriples.get(t.getPredicate().getURI()).add(
-                                    substringBeforeLast(t.getObject().getURI(), "/" + JCR_CONTENT));
-                            } else if (t.getObject().isLiteral()) {
-                                aclTriples.get(t.getPredicate().getURI()).add(
-                                    t.getObject().getLiteralValue().toString());
+                    child.getTriples(translator, PROPERTIES).filter(hasAclPredicate)
+                        .forEach(triple -> {
+                            aclTriples.putIfAbsent(triple.getPredicate().getURI(), new ArrayList<>());
```

Comment:
No need to do this and then use `get` below, use [`computeIfAbsent`](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#computeIfAbsent-K-java.util.function.Function-), which always returns a value. So `List<String> things = aclTriples.computeIfAbsent…; things.add(); things.add(); `